quickwit/quickwit-control-plane/src/indexing_scheduler/scheduling/mod.rs

```diff
@@ -238,8 +255,99 @@ pub enum SourceToScheduleType {
     IngestV1,
 }
 
+fn group_shards_into_pipelines(
+    source_uid: &SourceUid,
+    shard_ids: &[ShardId],
+    previous_indexing_tasks: &[IndexingTask],
+    load_per_shard: Load,
+) -> Vec<IndexingTask> {
+    let num_shards = shard_ids.len() as u32;
+    if num_shards == 0 {
+        return Vec::new();
+    }
+    let max_num_shards_per_pipeline = MAX_LOAD_PER_PIPELINE / load_per_shard;
+
+    // We compute the number of pipelines we will create, cooking in some hysteresis effect here.
+    // We have two different threshold to increase and to decrease the number of pipelines.
+    let min_num_pipelines: u32 =
```

Comment:
Can we protect the code from a division by 0? It's hard to check in the code if `load_per_shard` will always stay under `MAX_LOAD_PER_PIPELINE` (for a Kafka source, it seems possible to have a load higher than `MAX_LOAD_PER_PIPELINE`).

Reply:
very good point. FYI in the design this is loosely enforced by forcing a shard to deliver at most `5MB/s`...

But even if the assumption that we can always index  faster than `5MB/s`, we will have to catch with some shards...